### PR TITLE
Pypi release

### PR DESCRIPTION
**Pull request by tcztzy**
I tested install with `open-webui`, would not affect `open-webui`-installed files, and put files into `<site-packages>/open_webui/pipelines`.

Next step: update `open-webui` to aware this package's installation and start a subprocess to serve it.

NOTE: pipeline should be io bounded and not CPU/GPU bounded, and depends on `open-webui`'s main loop, so subprocess is ok.

### CODE REVIEW

**Comment by tjbck** [comment]
LGTM, is there a way to set the version externally from another file?

File: open_webui/pipelines/main.py [hunk 1]
```diff
@@ -231,3 +210,41 @@ def stream_content():
...
+
+        pipeline = loaded_module.Pipeline()
+
+        PIPELINES[loaded_module.__name__] = {
+            "module": pipeline,
+            "id": pipeline.id if hasattr(pipeline, "id") else loaded_module.__name__,
+            "name": pipeline.name if hasattr(pipeline, "name") else loaded_module.__name__,
+        }
+
+cli = typer.Typer(callback=callback)
```
Comment (tjbck) [inline open_webui/pipelines/main.py:230]: Ideally we'd want to keep typer separate from the main.py like our main open webui repo.

**Comment by tcztzy** [comment]
> LGTM, is there a way to set the version externally from another file?

hatch support using extract version from file with regex, just like what I do in [open-webui](https://github.com/open-webui/open-webui/blob/9d4c07b76a1682a24223dd0ae53ac6657bea3a90/pyproject.toml#L98)

File: open_webui/pipelines/main.py (hunk 1 above)
Comment (tcztzy) [reply open_webui/pipelines/main.py:230]: > Ideally we'd want to keep typer separate from the main.py like our main open webui repo.

OK, I will put typer in another file

**Comment by tjbck** [comment]
Closing due to conflicts but I encourage you to reopen! 🙌

**Comment by tcztzy** [comment]
If `pipelines` needs install by pypi along with `open-webui`, name conflict must be solved, such as `main.py`/`config.py`/`util`.

**Pull request closed without merging**